### ISSUE TITLE
On inser errort: Can't overwrite elasticsearch=6010499 with 6050199

### ISSUE THREAD

**Issue by flascaray** [crate/crate#8228]
**CrateDB version**: 3.2.3

**Environment description**:  Docker version 18.06.1-ce, build 0ffa825 

**Problem description**:

When insert data with a new attribute to my object (dynamic) CrateDB return the following error:

```JSON
{
    "error": {
        "message": "SQLActionException[SQLParseException: Can't overwrite elasticsearch=6010499 with 6050199]",
        "code": 4000
    }
}
```
```sh
curl -X POST \
  http://xxxxxxx:4200/_sql \
  -H 'Content-Type: application/json' \
  -d '{
  "stmt": "INSERT INTO moba.user_events (message_id, origin_ip, created, country, event_id, account_id, location, meta, seeker_id, provider_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
  "args": [
    "3782572971220403",
    "190.210.237.232",
    "2019-02-11T16:24:20Z",
    "AR",
    "SEEKER_PROFILE_VISITED5",
    "WXGdPKYAkFazMBo5u1HVK775r2",
    [-34.6029076, -58.3899023],
    {
      "positionId": "1",
      "positionId2": "2"
    },
    null,
    "25a3df4b69404a223fb090e881f64e7700"
  ]
}'
```
```sql
CREATE TABLE IF NOT EXISTS "moba"."user_events" (
   "account_id" STRING NOT NULL,
   "country" STRING NOT NULL,
   "created" TIMESTAMP NOT NULL,
   "created_month" STRING GENERATED ALWAYS AS CAST(date_trunc('month', "created") AS string),
   "event_id" STRING NOT NULL,
   "location" GEO_POINT,
   "message_id" STRING NOT NULL,
   "meta" OBJECT (DYNAMIC) AS (
      "customPosition" STRING,
      "gender_id" LONG,
      "payment" LONG,
      "paymentUnitId" STRING,
      "positionId" STRING,
      "positionOrder" LONG,
      "relatedPositions" BOOLEAN,
      "seekerPositionId" STRING
   ),
   "origin_ip" STRING,
   "provider_id" STRING,
   "seeker_id" STRING
)
CLUSTERED BY ("account_id") INTO 5 SHARDS
PARTITIONED BY ("country", "event_id", "created_month")
WITH (
   "allocation.max_retries" = 5,
   "blocks.metadata" = false,
   "blocks.read" = false,
   "blocks.read_only" = false,
   "blocks.read_only_allow_delete" = false,
   "blocks.write" = false,
   column_policy = 'dynamic',
   "mapping.total_fields.limit" = 1000,
   number_of_replicas = '0-all',
   refresh_interval = 10000,
   "routing.allocation.enable" = 'all',
   "routing.allocation.total_shards_per_node" = -1,
   "translog.durability" = 'ASYNC',
   "translog.flush_threshold_size" = 536870912,
   "translog.sync_interval" = 5000,
   "unassigned.node_left.delayed_timeout" = 60000,
   "warmer.enabled" = true,
   "write.wait_for_active_shards" = 'ALL'
)
```

The problem is with the *meta* field. It would seem that new fields can not be added. 

With this meta works
```json
{
      "positionId": "1"
}
```
With this meta does not work
```json
{
      "positionId": "1",
      "positionId2": "2"
}
```

**Comment by flascaray** [comment]
I drop the table and recreate it and works. 

Could something happen when upgrading from version 3.1 to version 3.2? The table was created in version 3.1 and I updated the docker image a few days ago

**Comment by mfussenegger** [comment]
Looks like there was an issue with the upgrade. We'll have a closer look at this.

**Comment by mfussenegger** [comment]
Thank you for the report. This was a quite serious regression. We've a fix pending for the next 3.2 release (https://github.com/crate/crate/pull/8236)
